examples/src/test/scala/akka/projection/examples/KafkaToSlickIntegrationSpec.scala

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.examples
+
+import scala.collection.immutable
+import scala.collection.immutable.Seq
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.kafka.ConsumerSettings
+import akka.kafka.scaladsl.Producer
+import akka.projection.ProjectionId
+import akka.projection.internal.MergeableOffsets
+import akka.projection.kafka.KafkaSourceProvider
+import akka.projection.kafka.KafkaSpecBase
+import akka.projection.scaladsl.SourceProvider
+import akka.projection.slick.SlickHandler
+import akka.projection.slick.SlickProjection
+import akka.projection.slick.SlickProjectionSpec
+import akka.projection.slick.internal.SlickOffsetStore
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.StringDeserializer
+import slick.basic.DatabaseConfig
+import slick.jdbc.H2Profile
+
+object KafkaToSlickIntegrationSpec {
+  object EventType {
+    val Login = "Login"
+    val Search = "Search"
+    val AddToCart = "AddToCart"
+    val CheckoutCart = "CheckoutCart"
+    val Logout = "Logout"
+  }
+
+  final case class UserEvent(userId: String, eventType: String)
+  final case class UserEventCount(eventType: String, count: Long)
+
+  val user1 = "user-id-1"
+  val user2 = "user-id-2"
+  val user3 = "user-id-3"
+
+  val userEvents = Seq(
+    UserEvent(user1, EventType.Login),
+    UserEvent(user1, EventType.Search),
+    UserEvent(user1, EventType.Logout),
+    UserEvent(user2, EventType.Login),
+    UserEvent(user2, EventType.Search),
+    UserEvent(user2, EventType.AddToCart),
+    UserEvent(user2, EventType.Logout),
+    UserEvent(user3, EventType.Login),
+    UserEvent(user3, EventType.Search),
+    UserEvent(user3, EventType.AddToCart),
+    UserEvent(user3, EventType.Search),
+    UserEvent(user3, EventType.AddToCart),
+    UserEvent(user3, EventType.CheckoutCart),
+    UserEvent(user3, EventType.Logout))
+
+  class EventTypeCountRepository(val dbConfig: DatabaseConfig[H2Profile]) {
+
+    import dbConfig.profile.api._
+
+    private class UserEventCountTable(tag: Tag) extends Table[UserEventCount](tag, "EVENTS_TYPE_COUNT") {
+      def eventType = column[String]("EVENT_TYPE", O.PrimaryKey)
+      def count = column[Long]("COUNT")
+      def * = (eventType, count).mapTo[UserEventCount]
+    }
+
+    def incrementCount(eventType: String)(implicit ec: ExecutionContext) = {
+      for {
+        count <- findByEventType(eventType).map {
+          case Some(userEventCount) => userEventCount.copy(count = userEventCount.count + 1)
+          case _                    => UserEventCount(eventType, 1)
+        }
+        _ <- userEventCountTable.insertOrUpdate(count)
```

Comment:
so the findByEventType and insertOrUpdate are atomic in same transaction? 
Is the count read by findByEventType guaranteed to not be updated by another EventTypeCountRepository instance inbetween the read and the insertOrUpdate?
Is all of it translated to sql running in the DB, or is it some DB lock, or how does that work?

Reply:
Good point.  I don't know how that works.  I doubt read and writes would be automatically handled in one transaction. I was following the same pattern used in `SlickProjectionImplSpec`, but this breaks down when more than 1 projection is updating some projection source with a different partition/shard/slice. /cc @renatocaval 

I'll need to rethink this example.

Reply:
What I would use for CRUD, like this, is optimistic locking. A version number column that is read in findByEventType and then used as condition in the update to be the same as what was read. The update also increments it by 1. If no row was updated it means that someone else updated inbetween. Then we could fail, and even use the retry mechanism from #95 

Would be pretty nice to include that pattern in this example.

An alternative would be to have one counter per projectionId, so that it's a single writer.

Reply:
added issue https://github.com/akka/akka-projection/issues/107 for this

Reply:
Reads and writes will be run in same transaction, but the issue here is not at Slick level, but the configured transaction isolation.

Reply:
To be more precise with what a mean, this is a general transaction issue and users need to configure the DB they way they want/need it and code accordingly to it. 

To have full protection here we will need either some optimistic locking or use a strict transaction isolation level (eg: Serializable). I don't think it's worth doing that in our tests unless we want to use it as a documentation sample.

Reply:
For the purpose of this example I was thinking of just querying to see if a count for an event type exists (instead of retrieving the count value itself) and then do either of the following:

- If it exists, issue an update statement and increment the count in place, instead of first reading it and incrementing it in code. i.e. `UPDATE EVENTS_TYPE_COUNT		 SET COUNT = COUNT + 1 WHERE EVENT_TYPE = ?`
- If it does not exist, insert the the record.  If the insert fails because another projection created it already then an update will happen on the retry. An alternative would just be to seed the records with the projection table DDL, but as @patriknw mentioned it would be a nice way to showcase retry logic.